Update lesson

jlboba opened the pull request:
Changes made: 

- Configured for TS 
- Added instructions for interfaces, typing 

Solution branch available

jlboba committed:
configure with ts

jlboba committed:
update lab with interfaces

jlboba committed:
fix instructions

lizbur10 reviewed (changes requested):
I recommend some resequencing of the instructions so that they clear the TypeScript errors for ProjectList and ProjectItem before they begin working on building out the components. They can't really see what's being rendered until those errors have been cleared.

Also the structure for the interfaces they need to build - in particular for ProjectItem - is pretty complex. I like that you're telling them to look at the logged object to see the structure, but there's still a bit of a leap to getting the interface set up correctly with the projects key pointing to an array of objects. Maybe a reference back to where that was covered in the lessons?

In general, there are a lot of layers here and they may need some additional help separating them all out.

jlboba committed:
add note about errors and how to close them

jlboba committed:
add hint on typing

jlboba committed:
fix note on closing errors

jlboba commented:
The errors are able to be closed out, I believe I showed them this in a previous lesson/lab. I can't remember anymore if it was a lesson prior to this or after though. So I just went ahead and added a paragraph on how to close out the errors so they can still see their work as they go through. 

Added a hint on typing the interface by showing a different example. 

Going to merge it in and delete the branch, but if there are still parts that could be clarified more, feel free to let me know!
